src/swarm-engine/Swarm/Game/CESK.hs

```diff
@@ -123,6 +123,13 @@ data Frame
     -- application) in environment @e@.  We will also push an 'FApp'
     -- frame on the stack.
     FArg Term Env
+  | -- | @FVArg v@ says that we were evaluating the left-hand side of
+    --   an application, and the next thing we should do is apply it
+    --   to the given value.  This does not normally occur as part of
+    --   the usual evaluation process for applications, which instead
+    --   uses FArg.  However, it is sometimes useful when reducing
+    --   other constructs.
+    FVArg Value
```

Comment:
What constructs?

Reply:
Heh, right now it's only `match`... but I think I remember other cases where `FVArg` might have been useful in the past.  I will make the comment more specific.